lib/bridge.js

```diff
@@ -17,51 +18,111 @@ var Callback = require('./callback');
  */
 var Bridge = function Bridge (ip, port, token, options) {
   if (!(this instanceof Bridge)) {
-    return new Bridge(ip, port, token);
+    return new Bridge(ip, port, token, options);
   }
 
   this.ip = ip;
   this.port = parseInt(port, 10);
   this.token = token;
+  this.type = 1; // 1 = hardware, 2 = software
+  
   this.delayBetweenRequests = options && options.delayBetweenRequests || 250;
   this.delayer = Promise.resolve();
 
   if (!this.ip || !this.port || !this.token || this.port < 1 || this.port > 65536) {
     throw new Error('Please check the arguments!');
   }
+  
+  // get bridge type
+  this._getBridgeType();
+};
+
+/**
+ * This function retrieves the type of bridge (1 = hardware, 2 = software).
+ *
+ * @param void
+ * @returns void
+ * @private
+ */
+Bridge.prototype._getBridgeType = function _getBridgeType () {
+
+	var self = this;
+	self.delayer = new Promise(resolve =>
+	{
+		self._request('info')
+			.then(bridge =>
+			{
+				console.log(JSON.stringify(bridge));
+				self.type = bridge.bridgeType;
+			})
+			.catch(err => self.type = 2)
+			.finally(() => resolve(true));
+	});
+};
+
+/**
+ * This function gets the required parameters for the token.
+ *
+ * @param {Boolean}		[forcePlainToken=false]		force to use plain instead of hashed token
+ * @returns {Object}
+ * @private
+ */
+Bridge.prototype._getTokenParams = function _getTokenParams (forcePlainToken) {
+
+  if (forcePlainToken || this.type != 1)
+    return { token: this.token };
+
+  else
+  {
+    var ts = new Date().toISOString().substr(0, 19)+'Z'; // YYY-MM-DDTHH:MM:SSZ
+	var rnr = Math.floor(Math.random() * 900000 + 100000); // 6 random digits
+	var hash = shajs('sha256').update(ts + ',' + rnr + ',' + this.token).digest('hex');
```

Comment:
Seems like the identation is wrong here. Mixing tabs and spaces. Please format the code in the same style as the already existing one